core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -116,6 +117,38 @@ public ThisT scanManifestsWith(ExecutorService executorService) {
     return self();
   }
 
+  @Override
+  public ThisT toBranch(String branch) {
+    throw new UnsupportedOperationException("Performing operations on a branch is currently not supported");
```

Comment:
@dimas-b , In the PR i have implemented the `toBranch` only in one of the subclasses of Snapshot producer aka FastAppend class. This is done in order to avoid creating a huge PR and be thorough in testing each of the child operations of snapshot producer. In future my aim is to incrementally raise PR for each of them.

Reply:
Thanks for the clarification, @namrathamyske ! Would you mind updating the message to something like `Performing [SnapshotProducer operations on a branch is currently not supported` to avoid making a (false) impression that it is not supported in "general"?.. or use different wording as you see fit. My only concern is that getting the old message in one context might be interpreted that it is not supported anywhere.